stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -748,10 +749,27 @@ static_assert(sizeof(ActiveTaskStatus) == ACTIVE_TASK_STATUS_SIZE,
 /// 1024-byte malloc quantum. We subtract off the slab header size, plus a
 /// little extra to stay within our limits even when there's overhead from
 /// malloc stack logging.
-static constexpr size_t SlabCapacity = 1024 - StackAllocator<0, nullptr>::slabHeaderSize() - 8;
+static constexpr size_t SlabCapacity = 1024 - StackAllocator<0, nullptr, nullptr>::slabHeaderSize() - 8;
 extern Metadata TaskAllocatorSlabMetadata;
 
-using TaskAllocator = StackAllocator<SlabCapacity, &TaskAllocatorSlabMetadata>;
+static bool disableTaskSlabAllocator() {
+  enum class EnableState: uint8_t {
+    Uninitialized,
+    Enabled,
+    Disabled,
+  };
+  static std::atomic<EnableState> savedState = { EnableState::Uninitialized };
```

Comment:
Is there a perf win with the tribool over a `std::atomic<std::optional<bool>>`?

Reply:
Probably not. It annoys me to use two bytes where one will suffice, though.